File: BTS/Java/Premiere annee/fonderie/Barre.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package fonderie;

/**
 *
 * @author therbron
 */
public class Barre {
    
    protected double densite;
    protected double longueur;
    protected String reference;
    
    Barre()
    {
        densite = 0.1f;
        longueur = 5;
        reference = "0001";
        System.out.println("Constructeur defaut Barre");
    }
    
    Barre(double densite, double longueur, String reference)
    {
        this.densite = densite;
        this.longueur = longueur;
        this.reference = reference;
        System.out.println("Constructeur parametre Barre");
    }
    
    Barre(Barre autreBarre)
    {
        densite = autreBarre.densite;
        longueur = autreBarre.longueur;
        reference = autreBarre.reference;
        System.out.println("Constructeur copie Barre");
    }
    
    void afficherReference()
    {
        System.out.println("Referenfe = " + reference);
    }
    
}

File: BTS/C++/polymorphismeCours/Barre.cpp
#include "Barre.h"

Barre::Barre(const float _densite, const float _longueur, const string _referenceBar)
    : densite(_densite), longueur(_longueur), referenceBarre(_referenceBar)
{
    cout << "Je suis le constructeur a argument de Barre." << endl;
}

Barre::Barre()
    : densite(5), longueur(2), referenceBarre("none")
{
    cout << "Je suis le constructeur par defaut de Barre." << endl;
}

Barre::Barre(const Barre &barre)
    : densite(barre.densite), longueur(barre.longueur), referenceBarre(barre.referenceBarre)
{
    cout << "Je suis le constructeur par copie de Barre." << endl;
}

Barre::~Barre()
{
    cout << "Barre detruite." << endl;
}

void Barre::afficherReference()
{
    cout << "Reference :" << referenceBarre << endl;
}
